### PR TITLE
Fix map

### PR DESCRIPTION
**Pull request by zkid18**
## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [x] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/contrib/datasets/movie_lens.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import os
```
Comment (Scitator) [inline catalyst/contrib/datasets/movie_lens.py]: do we need it?

File: catalyst/utils/metrics/hitrate.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""
```
Comment (Scitator) [inline catalyst/utils/metrics/hitrate.py]: do we need this file?

File: catalyst/utils/metrics/tests/test_hitrate.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import torch
```
Comment (Scitator) [inline catalyst/utils/metrics/tests/test_hitrate.py]: and this one?

**Review by Scitator — DISMISSED** [review]

File: catalyst/metrics/tests/test_avg_precision.py [hunk 4]
```diff
@@ -97,6 +105,30 @@ def test_mean_avg_precision():
...
+
+    map_k = metrics.mean_avg_precision(y_pred_torch, y_true_torch, top_k)
+
+    map_at1 = map_k[0]
+    map_at3 = map_k[1]
+    map_at5 = map_k[2]
+    map_at10 = map_k[3]
+
+    assert np.allclose(map_at1, 0.5, atol=1e-3)
+    assert np.allclose(map_at3, 0.667, atol=1e-3)
```
Comment (Scitator) [inline catalyst/metrics/tests/test_avg_precision.py]:
```suggestion
    assert np.allclose(map_at3, 0.6675, atol=1e-3)
```

File: catalyst/metrics/tests/test_avg_precision.py [hunk 5]
```diff
@@ -97,6 +105,30 @@ def test_mean_avg_precision():
...
+    assert np.allclose(map_at5, 0.6417, atol=1e-3)
```
Comment (Scitator) [inline catalyst/metrics/tests/test_avg_precision.py]:
```suggestion
    assert np.allclose(map_at5, 0.6425, atol=1e-3)
```

File: catalyst/metrics/tests/test_avg_precision.py [hunk 6]
```diff
@@ -97,6 +105,30 @@ def test_mean_avg_precision():
...
+
+    top_k = [1, 3, 5, 10]
+
+    map_k = metrics.mean_avg_precision(y_pred_torch, y_true_torch, top_k)
+
+    map_at1 = map_k[0]
+    map_at3 = map_k[1]
+    map_at5 = map_k[2]
+    map_at10 = map_k[3]
+
```
Comment (Scitator) [inline catalyst/metrics/tests/test_avg_precision.py]: let's add the comments like:
```
01: (1.0 + 0.0) / 2 = 0.5
03: ((1 + 0.67)/2 + 0.5) / 2 = 0.6675
05: ((1 + 0.67)/2 + (0.5 + 0.4)/2) / 2 = 0.6425
10: ((1 + 0.67 + 0.5 + 0.44 + 0.5)/5 + (0.5 + 0.4 + 0.43)/3 ) / 2  = 0.53
```

**Review by Scitator — DISMISSED** [review]

**Comment by zkid18** [comment]
Close this PR in the regard of https://github.com/catalyst-team/catalyst/pull/1140

**Pull request closed without merging**